lib/redis_client/cluster/router.rb

```diff
@@ -221,6 +223,26 @@ def close
         @node.each(&:close)
       end
 
+      def with(key:, write: true, retry_count: 0)
```

Comment:
[IMO] It looks like this method has two responsibilities. Those are finding a node and yielding a block. I'd say that it would be better to dedicate only finding a node in this method and to call `#try_delegate` by `RedisClient::Cluster` side.

Reply:
That seems fair enough. I pushed up a new version without this method at all.

* I wrote a new method `find_node_key_by_key` (it's a bit wordy, but does what it says) and re-implemented `find_node_key` and `find_primary_node_key` in terms of that.
* `RedisClient::Cluster` calls `@router.find_node_key_by_key`, then `@router.find_node`, and finally `@router.try_delegate` itself.